Cesium3DTiles/src/upgradeBatchTableToFeatureMetadata.cpp

```diff
@@ -0,0 +1,1248 @@
+    } else if (it->IsInt64()) {
+      int64_t value = it->GetInt64();
+      if (isInRangeForSignedInteger<int8_t>(value) &&
+          result.type <= PropertyType::Int8) {
+        result.type = PropertyType::Int8;
+      } else if (
+          isInRangeForSignedInteger<uint8_t>(value) &&
+          result.type <= PropertyType::Uint8) {
+        result.type = PropertyType::Uint8;
+      } else if (
+          isInRangeForSignedInteger<int16_t>(value) &&
+          result.type <= PropertyType::Int16) {
+        result.type = PropertyType::Int16;
+      } else if (
+          isInRangeForSignedInteger<uint16_t>(value) &&
+          result.type <= PropertyType::Uint16) {
+        result.type = PropertyType::Uint16;
+      } else if (
+          isInRangeForSignedInteger<int32_t>(value) &&
+          result.type <= PropertyType::Int32) {
+        result.type = PropertyType::Int32;
+      } else if (
+          isInRangeForSignedInteger<uint32_t>(value) &&
+          result.type <= PropertyType::Uint32) {
+        result.type = PropertyType::Uint32;
+      } else if (result.type <= PropertyType::Int64) {
+        result.type = PropertyType::Int64;
+      }
+    } else if (it->IsUint64()) {
+      result.type = PropertyType::Uint64;
+    } else if (it->IsLosslessFloat()) {
+      result.type = PropertyType::Float32;
+    } else if (it->IsDouble()) {
+      result.type = PropertyType::Float64;
+    } else if (
+        it->IsArray() && !it->Begin()->IsArray() &&
+        (result.type == PropertyType::None ||
+         result.type == PropertyType::Array)) {
+      auto memberResult = findCompatibleTypes(*it);
+
+      result.type = PropertyType::Array;
+      if (!result.componentType || result.componentType < memberResult.type) {
+        result.componentType = memberResult.type;
+      }
+
+      result.maxComponentCount =
+          result.maxComponentCount
+              ? glm::max(*result.maxComponentCount, it->Size())
+              : it->Size();
+      result.minComponentCount =
+          result.minComponentCount
+              ? glm::min(*result.minComponentCount, it->Size())
+              : it->Size();
+    } else {
+      // A string, null, or something else. So convert to string
+      result = CompatibleTypes{};
+      result.type = PropertyType::String;
+      break;
+    }
+  }
+
+  return result;
+}
+
+void updateExtensionWithJsonStringProperty(
+    Model& gltf,
+    ClassProperty& classProperty,
+    FeatureTable& /*featureTable*/,
+    FeatureTableProperty& featureTableProperty,
+    const rapidjson::Value& propertyValue) {
+  uint64_t totalSize = 0;
+  std::vector<rapidjson::StringBuffer> rapidjsonStrBuffers;
+  rapidjsonStrBuffers.reserve(propertyValue.Size());
+  for (const auto& v : propertyValue.GetArray()) {
+    rapidjson::StringBuffer& buffer = rapidjsonStrBuffers.emplace_back();
+    rapidjson::Writer<rapidjson::StringBuffer> writer(buffer);
+    v.Accept(writer);
+    totalSize += buffer.GetSize();
```

Comment:
I updated to use one buffer in the new commit. But for the string alone, I have to add the string manually to the buffer because the json writer adds double quotes which is not needed by us